packages/agreement-process/src/services/agreementSubmissionProcessor.ts

```diff
@@ -0,0 +1,499 @@
+): Promise<void> => {
+  const consumer = await tenantQuery.getTenantById(agreement.consumerId);
+
+  if (
+    !consumer?.data.mails.find(
+      (mail) => mail.kind === tenantMailKind.ContactEmail
+    )
+  ) {
+    throw consumerWithNotValidEmail(agreement.id, agreement.consumerId);
+  }
+};
+
+const getUpdateSeed = (
+  descriptor: Descriptor,
+  consumer: Tenant,
+  eService: EService,
+  agreement: Agreement,
+  payload: ApiAgreementSubmissionPayload,
+  stamps: AgreementStamps,
+  newState: AgreementState,
+  suspendedByPlatform: boolean
+): UpdateAgreementSeed =>
+  newState === agreementState.active
+    ? {
+        state: newState,
+        certifiedAttributes: matchingCertifiedAttributes(descriptor, consumer),
+        declaredAttributes: matchingDeclaredAttributes(descriptor, consumer),
+        verifiedAttributes: matchingVerifiedAttributes(
+          eService,
+          descriptor,
+          consumer
+        ),
+        suspendedByConsumer: agreement.suspendedByConsumer,
+        suspendedByProducer: agreement.suspendedByProducer,
+        suspendedByPlatform,
+        consumerNotes: payload.consumerNotes,
+        stamps,
+      }
+    : {
+        state: newState,
+        certifiedAttributes: [],
+        declaredAttributes: [],
+        verifiedAttributes: [],
+        suspendedByConsumer: undefined,
+        suspendedByProducer: undefined,
+        suspendedByPlatform,
+        consumerNotes: payload.consumerNotes,
+        stamps,
+      };
+
+const validateResultState = (
+  state: AgreementState,
+  agreementId: string
+): void => {
+  if (state !== agreementState.active && state !== agreementState.pending) {
+    throw agreementSubmissionFailed(agreementId);
+  }
+};
+
+const suspendedByPlatformFlag = (fsmState: AgreementState): boolean =>
+  fsmState === agreementState.suspended ||
+  fsmState === agreementState.missingCertifiedAttributes;
+
+const isActiveOrSuspended = (state: AgreementState): boolean =>
+  state === agreementState.active || state === agreementState.suspended;
+
+const calculateStamps = (
+  agreement: Agreement,
+  state: AgreementState,
+  stamp: AgreementStamp
+): AgreementStamps =>
+  match<AgreementState, AgreementStamps>(state)
+    .with(agreementState.draft, () => agreement.stamps)
+    .with(agreementState.pending, () => ({
+      ...agreement.stamps,
+      submission: stamp,
+    }))
+    .with(agreementState.active, () => ({
+      ...agreement.stamps,
+      submission: stamp,
+      activation: stamp,
+    }))
+    .with(agreementState.missingCertifiedAttributes, () => agreement.stamps)
+    .otherwise(() => {
+      throw agreementNotInExpectedState(agreement.id, state);
+    });
+
+/* 
+  This function emulate javax.mail.internet.InternetAddress parse on list of email addresses 
+  https://docs.oracle.com/javaee/6/api/javax/mail/internet/InternetAddress.html#parse(java.lang.String,%20boolean)
+*/
+export const extractEmailAddresses = (input: string): InternetAddress[] => {
```

Comment:
there is no need to validate the email addresses.
In Scala the check is done because the library expects a specific class for the mail, and parsing is the only way to create the instance